File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        ans = 0
        currCount = 0
        chk = defaultdict(int)
        chk[0] = 1

        for i in range(0,len(nums)):
            if nums[i] % 2 != 0:
                currCount += 1

            term = currCount - k 
            if term in chk:
                ans += chk[term]
            chk[currCount] += 1

        return ans
           

        

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) 
    {
        int ans = 0;
        int oddCount = 0;
        
        map<int,int> freq;
        freq[0] = 1;

        for(int i=0 ; i < nums.size() ;i++)
        {
            if (nums[i] % 2 != 0)
            {
                oddCount += 1;
            }
            int term = oddCount - k;

            if (freq.find(term) != freq.end())
            {
                ans += freq[term];
            }

            freq[oddCount]++;
        }

        return ans;
        
    }
};